pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/server/KinesisDataServerStartable.java

```diff
@@ -0,0 +1,120 @@
+import software.amazon.awssdk.http.SdkHttpConfigurationOption;
+import software.amazon.awssdk.http.apache.ApacheSdkHttpService;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
+import software.amazon.awssdk.services.kinesis.model.CreateStreamRequest;
+import software.amazon.awssdk.services.kinesis.model.DescribeStreamRequest;
+import software.amazon.awssdk.utils.AttributeMap;
+
+
+public class KinesisDataServerStartable implements StreamDataServerStartable {
+  private static final Logger LOGGER = LoggerFactory.getLogger(KinesisDataServerStartable.class);
+
+  public static final String NUM_SHARDS_PROPERTY = "numShards";
+  public static final String DEFAULT_REGION = "us-east-1";
+  public static final String DEFAULT_ACCESS_KEY = "access";
+  public static final String DEFAULT_SECRET_KEY = "secret";
+  public static final String DEFAULT_PORT = "4566";
+
+  private final Localstack _localstackDocker = Localstack.INSTANCE;
+  LocalstackDockerConfiguration _dockerConfig;
+  Properties _serverProperties;
+  private String _localStackKinesisEndpoint = "http://localhost:%s";
+
+  @Override
+  public void init(Properties props) {
+    _serverProperties = props;
+    final Map<String, String> environmentVariables = new HashMap<>();
+    environmentVariables.put("SERVICES", ServiceName.KINESIS);
+    _dockerConfig =
+        LocalstackDockerConfiguration.builder().portEdge(_serverProperties.getProperty("port", DEFAULT_PORT))
+            .portElasticSearch(String.valueOf(NetUtils.findOpenPort(4571))).imageTag("0.12.15")
+            .environmentVariables(environmentVariables).build();
+
+    _localStackKinesisEndpoint =
+        String.format(_localStackKinesisEndpoint, _serverProperties.getProperty("port", DEFAULT_PORT));
+  }
+
+  @Override
+  public void start() {
+    _localstackDocker.startup(_dockerConfig);
+  }
+
+  @Override
+  public void stop() {
+    _localstackDocker.stop();
+  }
+
+  @Override
+  public void createTopic(String topic, Properties topicProps) {
+    try {
+      KinesisClient kinesisClient = KinesisClient.builder().httpClient(
+              new ApacheSdkHttpService().createHttpClientBuilder().buildWithDefaults(
+                  AttributeMap.builder().put(SdkHttpConfigurationOption.TRUST_ALL_CERTIFICATES, Boolean.TRUE).build()))
+          .credentialsProvider(getLocalAWSCredentials()).region(Region.of(DEFAULT_REGION))
+          .endpointOverride(new URI(_localStackKinesisEndpoint)).build();
+
+      kinesisClient.createStream(
+          CreateStreamRequest.builder().streamName(topic).shardCount((Integer) topicProps.get(NUM_SHARDS_PROPERTY))
+              .build());
+
+      String kinesisStreamStatus =
+          kinesisClient.describeStream(DescribeStreamRequest.builder().streamName(topic).build()).streamDescription()
+              .streamStatusAsString();
+
+      while (!kinesisStreamStatus.contentEquals("ACTIVE")) {
```

Comment:
wrap this in a waitForCondition loop (like in TestUtils), so we're not stuck here forever in case of some issue

Reply:
done